ui/ui.go

```diff
@@ -42,14 +43,36 @@ func NewHandler(version string, us user.Service, cs config.Service, chs challeng
 	return &uiHandler{version, us, cs, chs, crs, as, store}
 }
 
-func (h *uiHandler) Route() *mux.Router {
+func (h *uiHandler) Route(unsafe bool) http.Handler {
+	key, err := h.configService.SessionKey()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	var CSRF func(http.Handler) http.Handler
+	if unsafe {
+		CSRF = csrf.Protect(
+			key,
+			csrf.SameSite(csrf.SameSiteStrictMode),
+			csrf.FieldName("csrf_token"),
+			csrf.Secure(false),
+		)
+	} else {
+		CSRF = csrf.Protect(
+			key,
+			csrf.SameSite(csrf.SameSiteStrictMode),
+			csrf.FieldName("csrf_token"),
+		)
+	}
+
```

Comment:
This could probably just be 
```
CSRF = csrf.Protect(
			key,
			csrf.SameSite(csrf.SameSiteStrictMode),
			csrf.FieldName("csrf_token"),
			csrf.Secure(!unsafe),
		)
```
Although I don't really know this library so it could be `csrf.Secure(true)` does something completely different

Reply:
And of course this point is moot if we expect there to be more configuration differences between safe and unsafe in the future.

Reply:
At first I was expecting more flags to be set but it turns out the defaults are all quite strong.